core/src/main/java/org/apache/iceberg/events/CreateSnapshotEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.events;
+
+import java.util.Map;
+
+public final class CreateSnapshotEvent implements UpdateEvent {
+  private final String operation;
+  private final String tablePath;
+  private final Map<String, String> summary;
+  private final Iterable<String> addedFiles;
+  private final Iterable<String> appendedManifests;
+  private final Iterable<String> deletePaths;
+
+  public CreateSnapshotEvent(
+      String operation, String tablePath, Map<String, String> summary,
+      Iterable<String> addedFiles, Iterable<String> appendedManifests, Iterable<String> deletePaths) {
```

Comment:
Why use table path/location instead of name? I think we should send a `name` field with whatever was used to load the table. We might need to add `name` to the `Table` interface to do that, but adding that method is easier than adding something to `TableOperations` because all the implementations are internal.

Reply:
I think we might need a reference to `Table` . Where we call commit and publish events we only have access to `TableOperations` . Any thoughts on how to get the `name` during `commits`?

Reply:
Table creates all of these operations, so it can pass in its name as well as `ops`. I'd prefer that to trying to parse a location.

Reply:
Maybe I can pass in the `name` as a property which can reside in table metadata. I can access that metadata during commit and I don't have to add any new api in `TableOperations` ?

Reply:
I don't think we need to add to `TableOperations`. Table keeps track of its name, and is responsible for creating the operations classes, like `FastAppend`. So we can just pass the name in along side `TableOperations`:

```java
  @Override
  public AppendFiles newFastAppend() {
    return new FastAppend(name, ops);
  }
```